components/camel-smb/src/main/java/org/apache/camel/component/smb/SmbProducer.java

```diff
@@ -0,0 +1,239 @@
+        String fileName = exchange.getIn().getHeader(Exchange.FILE_NAME, String.class);
+
+        SmbConfiguration configuration = getEndpoint().getConfiguration();
+        String path = (configuration.getPath() == null) ? "" : configuration.getPath();
+
+        LOGGER.debug("storeFile() fileName [" + fileName + "]");
+
+        try {
+            connectIfNecessary(exchange);
+
+            java.io.File file = new java.io.File(path + java.io.File.separator + fileName);
+
+            DiskShare share = (DiskShare) session.connectShare(getEndpoint().getShareName());
+            createDirectory(share, file);
+
+            GenericFileExist gfe = determineFileExist(exchange);
+
+            SMB2CreateDisposition scd = SMB2CreateDisposition.FILE_OPEN;
+            HashSet<AccessMask> accessMask = new HashSet<AccessMask>(Arrays.asList(AccessMask.GENERIC_ALL));
+            HashSet<SMB2CreateOptions> createOptions = new HashSet<>(Arrays.asList(SMB2CreateOptions.FILE_DIRECTORY_FILE));
+            HashSet<FileAttributes> fileAttributes
+                    = new HashSet<FileAttributes>(Arrays.asList(FileAttributes.FILE_ATTRIBUTE_NORMAL));
+
+            // File existence modes
+            switch (gfe) {
+                case Override:
+                    scd = SMB2CreateDisposition.FILE_OVERWRITE_IF;
+                    break;
+                case Fail:
+                    scd = SMB2CreateDisposition.FILE_CREATE;
+                    if (share.fileExists(file.getPath())) {
+                        doFail(file.getPath());
+                    }
+                    break;
+                case Append:
+                    scd = SMB2CreateDisposition.FILE_OPEN_IF;
+                    accessMask = new HashSet<AccessMask>(Arrays.asList(AccessMask.FILE_WRITE_DATA));
+                    break;
+                case Ignore:
+                    if (share.fileExists(file.getPath())) {
+                        doIgnore(file.getPath());
+                        return;
+                    }
+                    break;
+                case Move:
+                    throw new UnsupportedOperationException("Move is not implemented for this producer at the moment");
+                case TryRename:
+                    throw new UnsupportedOperationException("TryRename is not implemented for this producer at the moment");
+            }
+
+            File shareFile = share.openFile(file.getPath(),
+                    accessMask,
+                    fileAttributes,
+                    SMB2ShareAccess.ALL,
+                    scd,
+                    createOptions);
+
+            InputStream is = (exchange.getMessage(InputStream.class) == null)
+                    ? exchange.getMessage().getBody(InputStream.class) : exchange.getMessage(InputStream.class);
+
+            // In order to provide append option, we need to use offset / write with shareFile rather
+            // than with outputstream
+            int buffer = 2048;
```

Comment:
I am curious about the choice of this value. Should we make it configurable?

Reply:
Added it as a configuration option in SmbConfiguration